### ISSUE TITLE
blockquotes violate principle of uniformity

### ISSUE THREAD

**Issue by vassudanagunta** [commonmark/commonmark-spec#634]
> [principle of uniformity](https://spec.commonmark.org/0.29/#principle-of-uniformity): if a chunk of text has a certain meaning, it will continue to have the same meaning when put into a container block (such as a list item or blockquote).

The following:

```
List with sublist:

- a
  - b

Exact same lines put in a blockquote container block:

>- a
>  - b
```

is rendered by CommonMark 0.29  as

![bq0](https://user-images.githubusercontent.com/2830093/73654648-43fee280-465a-11ea-99e5-6a4a9e4abc03.png)

Please compare the above with the example given in the spec just below the definition of the [principle of uniformity](https://spec.commonmark.org/0.29/#principle-of-uniformity)

[A plurality of markdown implementations, including the original Gruber, correctly follow the principle](https://johnmacfarlane.net/babelmark2/?normalize=1&text=List+with+sublist%3A%0A%0A-+a%0A++-+b%0A%0AExact+same+lines+put+in+a+blockquote+container+block%3A%0A%0A%3E-+a%0A%3E++-+b) in this regard (I'm counting all the CommonMark implementations as one). 

Another example of the issue:

```
     indented 5 spaces
    indented 4 spaces
     indented 5 spaces

The above lines, in a blockquote:

>     indented 5 spaces
>    indented 4 spaces
>     indented 5 spaces
```
is rendered as:

![bq2](https://user-images.githubusercontent.com/2830093/73656885-0e102d00-465f-11ea-8b4a-963f3aebc18e.png)

Though [in this case, all the implementations except MultiMarkdown get it wrong](https://johnmacfarlane.net/babelmark2/?normalize=1&text=+++++indented+5+spaces%0A++++indented+4+spaces%0A+++++indented+5+spaces%0A%0AThe+above+lines%2C+in+a+blockquote%3A%0A%0A%3E+++++indented+5+spaces%0A%3E++++indented+4+spaces%0A%3E+++++indented+5+spaces%0A). I would say that this case is esoteric and unimportant, but the handling of nested lists as demonstrated in the first example, should determine the proper course of action.

**Comment by vassudanagunta** [comment]
This issue has common cause (double entendre intended) with #421 and #460, both of which are currently attached to the 0.30 milestone. 

I think the many proposals considered under #466 should also be evaluated against at least the first example above.

**Comment by Crissov** [comment]
The spec should really disallow blockquote markers without a single space following it, 

~~~~ markdown 
>- a
>  - b
~~~~

~~~~ html 
<p>&gt;- a
&gt;  - b</p>
~~~~

but since that would break backwards compatibility, they should tu least not be mend together in a single blockquote, cf. different bullet list markers.

Then, this Commonmark input 

~~~~ markdown 
>- a
>  - b
~~~~

would be identical to

~~~~ markdown 
>- a

>  - b
~~~~

and both would yield the same (normalized) HTML output:

~~~~ html 
<blockquote><ul><li>a</li></ul></blockquote>
<blockquote><ul> <li>b</li></ul></blockquote>
~~~~

**Comment by vassudanagunta** [comment]
That interpretation, @Crissov, to me is very counterintuitive and runs counter to Markdown's design principles.

**Comment by jgm** [comment]
Yeah, this is a problem for sure.  As noted in my comment on #460, it's a tricky one.

**Comment by Crissov** [comment]
@vassudanagunta Blockquote markers are the only line prefixes that do not require a space following them, since hash-sign headings have been fixed.
